samples/matter/smoke_co_alarm/src/app_task.cpp

```diff
@@ -0,0 +1,469 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "app_task.h"
+
+#include "app/matter_init.h"
+#include "app/task_executor.h"
+#include "board/board.h"
+#include "lib/core/CHIPError.h"
+#include "lib/support/CodeUtils.h"
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+#include "dfu/ota/ota_util.h"
+#endif
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/identify-server/identify-server.h>
+#include <app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerHandler.h>
+#include <app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.h>
+#include <app/server/OnboardingCodesUtil.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace ::chip;
+using namespace ::chip::app;
+using namespace ::chip::DeviceLayer;
+
+namespace
+{
+
+constexpr uint32_t kExpressedStateLedEvenBlinkMs = 300;
+constexpr uint32_t kExpressedStateLedOddBlink1Ms = 300;
+constexpr uint32_t kExpressedStateLedOddBlink2Ms = 700;
+constexpr uint32_t kSelfTestDurationMs = 5000;
+constexpr uint32_t kSelfTestLedUpdateTimeMs = 200;
+
+static std::array<Clusters::SmokeCoAlarm::ExpressedStateEnum, SmokeCoAlarmServer::kPriorityOrderLength>
+	sPriorityOrder = { Clusters::SmokeCoAlarm::ExpressedStateEnum::kSmokeAlarm,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kCOAlarm,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kHardwareFault,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kTesting,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kEndOfService,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kBatteryAlert,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kInterconnectSmoke,
+			   Clusters::SmokeCoAlarm::ExpressedStateEnum::kInterconnectCO };
+
+Identify sIdentify = { AppTask::Instance().kSmokeCoAlarmEndpointId, AppTask::IdentifyStartHandler,
+		       AppTask::IdentifyStopHandler, Clusters::Identify::IdentifyTypeEnum::kVisibleIndicator };
+
+#ifdef CONFIG_CHIP_ICD_UAT_SUPPORT
+#define UAT_BUTTON_MASK DK_BTN3_MSK
+#endif
+} /* namespace */
+
+void AppTask::IdentifyStartHandler(Identify *)
+{
+	Nrf::PostTask(
+		[] { Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).Blink(Nrf::LedConsts::kIdentifyBlinkRate_ms); });
+}
+
+void AppTask::IdentifyStopHandler(Identify *)
+{
+	Nrf::PostTask([] { Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).Set(false); });
+}
+
+void AppTask::ButtonEventHandler(Nrf::ButtonState state, Nrf::ButtonMask hasChanged)
+{
+	// if ((APPLICATION_BUTTON_MASK & state & hasChanged)) {
```

Comment:
This is a work in progress, right?

Reply:
No, I just forgot to remove it :smile: Thanks!